Fix for issue #90.

JGRennison opened the pull request:
Fix add_ts_packet_data calculating wrong section length when first chunk <3 bytes.
Delay expected length calculation until at least 3 bytes accumulated.
Delay section start logging until at least 8 bytes (full header) accumulated.
Fixes issue #90.

Commit:
Rewrite of autoconfiguration code : start of MuMuDVB V2

Autoconfiguration partial has been removed in favor of a more flexible
system. Big commit but complete rewrite difficult to split.
The new system allow the user to override any parameter

braice committed:
Rewrite of autoconfiguration code : start of MuMuDVB V2

Autoconfiguration partial has been removed in favor of a more flexible
system. Big commit but complete rewrite difficult to split.
The new system allow the user to override any parameter

andir committed:
Fixed compile error due to missing/additional bracket

braice committed:
Merge pull request #50 from andir/mumudvb2_fix_log_bracket

Fixed compile error due to missing/additional bracket

braice committed:
Fix compile

Commit:
Segfault when several channels share the same PMT fixed, compilation of documentation

OuNao committed:
Corrected handling of UTF-8 encoding by iconv.

Thanks to OuNao for finding and fixing the bug
https://github.com/OuNao

braice committed:
Fix autoconf when several channels share the same PMT PID

OuNao committed:
Corrected build for Android. Included --enable-android for build compatibility.

OuNao committed:
Included service_type 0xc0 for Brazilian 1seg tv transmissions.

braice committed:
Doc update

OuNao committed:
Build script for android.

braice committed:
Index page for unicast, close issue #54

braice committed:
Merge pull request #52 from OuNao/mumudvb2

Build compatibility for android

braice committed:
Documentation update

braice committed:
SDT autoconf: check table id
The version is only for a specific table id so we need to check both to know if we need update

braice committed:
Fix for new autoconf and CAM support

braice committed:
Better error messages when special bandwidth not supported

braice committed:
Cleanup, in progress

braice committed:
Cleanup, scam improvement, unicam own fds
 * To avoid confusion, the file descriptors for unicast are polled independantly
 * Global variables removed
 * Work on software CAM, the functions are called according to the new channel organization

braice committed:
Coverity fixes

sesse committed:
Stop using FE_GET_EVENT for tuning

FE GET EVENT will be deprecated in future kernel releases
Allow the user with the --enable-tune_old flag to fallback in case of issue

Signed-off-by: Brice Dubost <mumudvb@braice.net>

braice committed:
Streamed channel : log only ready channels

braice committed:
Tune : Skip event description when FE_GET_FRONTEND not supported

Commit:
Network ids: store and use

Store the network_id and the original_network_id (will be available in the webservices)
Use the network_id for NIT table parsing

Commit:
Monitor : state in JSON and information about autoconf

braice committed:
Details

braice committed:
Possibility to select Diseqc switch position 0 (closes #66)

braice committed:
SAP announces are not sent when LCN is asked and NIT not read

braice committed:
Log messages : channel_next does not exist anymore, replaced by new_channel

Commit:
Electronic Program Guide in the Webservices
use the store_eit option and /monitor/EIT.json

Commit:
Fix a possible infinite loop

Commit:
Better string handling for EIT to JSON

ostryck committed:
make back SCAM support operational

ostryck committed:
scam: store proper channels pmt in scam_pmt_packet

ostryck committed:
scam: change ring_buffer to 1d

ostryck committed:
bring back new key retrieval method

braice committed:
Documentation update

centove committed:
Add a client detail line to each channel for connected unicast clients

braice committed:
Option to enable/disable DiseQC messages repetition

Commit:
Possible segmentation fault in string conversion corrected

centove committed:
Merge remote-tracking branch 'upstream/mumudvb2' into mumudvb2
Pulling in updated tuning file

centove committed:
Fix JSON generation

centove committed:
Feh

centove committed:
Update the client_list output to a seperate function.
Clean up the json channel list output for easier reading/debugging.
Sync the json state and xml state - I think they are the same now.
Add the client list to the xml output.

centove committed:
Fix couple minor bugs

braice committed:
Merge pull request #72 from centove/mumudvb2

Add a client detail line to each channel for connected unicast clients

braice committed:
EIT: length check

centove committed:
Documentation update, and a small fix/standarization between the .json/.xml
output for the client list.

centove committed:
Stuipd stupid missing )

braice committed:
Merge pull request #75 from centove/mumudvb2

Documentation update and json for connected clients

braice committed:
Spell

braice committed:
Doc update

m0yellow committed:
experimental support of HEVC stream, tested with currently available SES
Test streams on Astra 19.2E 10994MHz

braice committed:
Merge pull request #69 from ostryck/mumudvb2

OSCAM back in MuMuDVB2 !

braice committed:
Merge pull request #77 from m0yellow/mumudvb2

Support HEVC (UHD) Channels

braice committed:
HD video type description

braice committed:
Better detection of null IP address

ostryck committed:
scam_getcw: locking fixes

braice committed:
Merge pull request #79 from ostryck/mumudvb2

scam_getcw: locking fixes (mumudvb2)

braice committed:
Better debug message

braice committed:
Merge branch 'mumudvb2' of github.com:braice/MuMuDVB into mumudvb2

braice committed:
Avoid crash when asking json on an empty transponder

braice committed:
Compile issues

braice committed:
Improved JSON for EIT/EPG

braice committed:
Documentation update

Happy new year !

lkarsten committed:
Add a basic .gitignore file.

Keep the clutter out of "git status".

lkarsten committed:
Correct some spelling mistakes.

Conflicts:
	doc/README_CONF.txt

lobodpav committed:
Implemented getting channel by name in HTTP unicasts, i.e. GET /byname/channelname

ubaldus committed:
Update unicast_EIT.c

parent rating descriptor can have more language/rating.

braice committed:
Merge pull request #87 from ubaldus/mumudvb2

Update unicast_EIT.c
Thanks !

braice committed:
Doc: Unicast streaming software

braice committed:
Merge branch 'mumudvb2' of github.com:braice/MuMuDVB into mumudvb2

braice committed:
EIT JSON more valid
improvement on the JSON validity: commas and unique identifiers

braice committed:
This is MuMuDVB 2

JGRennison committed:
EIT rewriter: Fix use of mumudvb_ts_packet_t full_buffer_len instead of len_full,
for the length of a packet in data_full.

Commit:
Fix for issue #95 - add ECM pids to streams with autconfiguration.

Commit:
revert change for idle time

braice commented on src/ts.c:
```diff
@@ -226,9 +250,21 @@ void add_ts_packet_data(unsigned char *buf, mumudvb_ts_packet_t *pkt, int data_l
+					pkt->pid,
+					pkt->cc,
+					pkt->len_partial);
+			return;
```
I would prefer a if() {A} - else() {B} structure rather than a structure if() {A return} B
Clearer to read

braice commented on src/ts.c:
```diff
@@ -293,6 +314,15 @@ void add_ts_packet_data(unsigned char *buf, mumudvb_ts_packet_t *pkt, int data_l
+			if(pkt->status_partial==PARTIAL_HEADER)
+			{
+				//read up to the first 3 bytes, to get the packet length
+				memcpy(pkt->data_partial+pkt->len_partial,buf,3-pkt->len_partial);
```
We need to check than (3-len_partial) > 0
And that we have enough data to read the length
Because of transmission errors, the TS data can be quite garbage sometimes

braice commented:
Overall it looks good, just see the two comments in the code and I'll include it
Thank you for your contribution

braice commented:
One last comment, the default branch is now mumuDVB2, I will drop the branch master as it is very soon

braice committed:
Merge pull request #98 from joeusercz/mumudvb2

Fix for issue #95 - add ECM pids to streams with autconfiguration.

braice committed:
Old option, fixes #97

JGRennison committed:
Fix section length calculation when first chunk less than 3 bytes.

This fix applies to section length calculation in add_ts_packet_data.
Delay expected length calculation until at least 3 bytes accumulated.
Delay section start logging until at least 8 bytes (full header)
accumulated.

JGRennison commented:
It's seems that it's not possible to change the target branch of a pull request, so I've rebased this into pull request #99 which is based on the mumudvb2 branch.
